pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+                        this.setState({ updating: false });
+                    });
+        }
+    }
+
+    render() {
+        // Not yet recognized
+        if (this.state.loaded === false || this.state.patchName === null)
+            return null;
+
+        // Not supported on this system
+        if (this.state.unavailable.length > 0)
+            return null;
+
+        let state = _("Enabled");
+        let actionText = _("Edit");
+        let action = () => this.setState({ showModal: true });
+
+        if (this.state.missing.length > 0 && this.state.unavailable.length === 0) {
+            state = _("Not installed");
+            actionText = _("Install");
+            action = this.handleInstall;
+        } else if (!this.state.enabled) {
+            state = _("Disabled");
+            actionText = _("Enable");
+        }
+
+        let kernel_name = "";
+        if (this.state.kernelName)
+            kernel_name = " (" + this.state.kernelName + ")";
+
+        let error = null;
+        if (this.state.error)
+            error = <Alert variant='danger' isInline title={this.state.error} />;
+
+        const body = <Checkbox id="apply-kpatch"
+                               isChecked={this.state.applyCheckbox}
+                               label={_("Apply kernel patches")}
+                               onChange={checked => this.setState({ applyCheckbox: checked })}
+                               body={<>
+                                   <Radio id="current-future" label={_("for current and future kernels")} onChange={() => this.setState({ justCurrent: false })} isDisabled={!this.state.applyCheckbox} isChecked={!this.state.justCurrent} />
+                                   <Radio id="current-only" label={_("for current kernel only") + kernel_name} onChange={() => this.setState({ justCurrent: true })} isDisabled={!this.state.applyCheckbox} isChecked={this.state.justCurrent} />
+                               </>}
+        />;
+
+        return (<>
+            <div id="kpatch-settings">
+                <Flex alignItems={{ default: 'alignItemsCenter' }}>
+                    <Flex grow={{ default: 'grow' }} alignItems={{ default: 'alignItemsBaseline' }}>
+                        <FlexItem>
+                            <b>{_("Kernel patching")}</b>
+                        </FlexItem>
+                        <FlexItem>
+                            {state}
+                        </FlexItem>
+                    </Flex>
+                    <Flex>
+                        <Button variant="secondary"
+                                isSmall
+                                disabled={!this.props.privileged}
+                                onClick={action}>
+                            {actionText}
+                        </Button>
+                    </Flex>
+                </Flex>
+            </div>
+            <Modal position="top" variant="small" id="kpatch-setup" isOpen={this.state.showModal}
```

Comment:
Does it make sense to even render the modal on `!showModal`? I thought our usual pattern was `{ show && <Modal isOpen ...> }`. Like this it would be okay as well, I just seem to remember that sometimes we ran into weird interactions with form submits.

Reply:
We are rather inconsistent about this. I am not aware of any issue when using this pattern and it feels a bit cleaner than having `{this.state.showModal && <Modal` pattern.